homeassistant/components/dsmr/config_flow.py

```diff
@@ -290,9 +308,14 @@ async def async_validate_dsmr(
         data = input_data
 
         try:
-            info = await _validate_dsmr_connection(self.hass, data)
+            try:
+                is_rfxtrx = False
+                info = await _validate_dsmr_connection(self.hass, data, is_rfxtrx)
+            except CannotCommunicate:
+                is_rfxtrx = True
+                info = await _validate_dsmr_connection(self.hass, data, is_rfxtrx)
 
-            data = {**data, **info}
+            data = {**data, **info, CONF_IS_RFXTRX: is_rfxtrx}
```

Comment:
Instead of adding an item just for rfxtrx, I suggest we name all the protocols and store which protocol to use. If another protocol is added we can then use the same data item.

Reply:
Yes, that makes sense.
Do you have suggestions for naming? Would just DSMR and RFXtrxDSMR work? Or does DSMRProtocol and RFXtrxDSMRProtocol (just like the names in the dsmr_parser library) make more sense?

Reply:
Those names sound ok.

Reply:
Updated code to use `CONF_PROTOCOL` parameter which is either `DSMR_PROTOCOL` or `RFXTRXDSMR_PROTOCOL`.